pkg/ec2wrapper/ec2wrapper.go

```diff
@@ -30,15 +30,15 @@ type EC2Wrapper struct {
 
 //NewMetricsClient returns an instance of the EC2 wrapper
 func NewMetricsClient() (*EC2Wrapper, error) {
-	metricsSession := session.Must(session.NewSession())
+	sess := awssession.New()
 	ec2MetadataClient := ec2metadatawrapper.New(nil)
 
 	instanceIdentityDocument, err := ec2MetadataClient.GetInstanceIdentityDocument()
 	if err != nil {
 		return &EC2Wrapper{}, err
 	}
 
-	ec2ServiceClient := ec2.New(metricsSession, aws.NewConfig().WithMaxRetries(maxRetries).WithRegion(instanceIdentityDocument.Region))
+	ec2ServiceClient := ec2.New(sess)
```

Comment:
Region info has been removed in the last patch and is not passed from elsewhere. You will need to configure it